js/video/src/StaticVideo.svelte

```diff
@@ -59,21 +63,37 @@
 			{label}
 		/>
 	{/key}
-	<div class="download" data-testid="download-div">
+	<div class="icon-buttons" data-testid="download-div">
 		<a
 			href={value.data}
 			target={window.__is_colab__ ? "_blank" : null}
 			download={value.orig_name || value.name}
 		>
 			<IconButton Icon={Download} label="Download" />
 		</a>
+		{#if shareable}
+			<IconButton
+				Icon={Community}
+				label="Share"
+				on:click={async () => {
+					if (!value) return;
+					let url = await uploadToHuggingFace(value.data, "url");
```

Comment:
I'm confused how this works 😅 

If you load an example video from the cache it will be base64 but the upload still seems to work even though we treat it as url?

From the video_component demo: 
<img width="1039" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/b86738c4-1d35-463a-81e3-37834aa0c3ee">

Reply:
Weird! Seems like you can `fetch()` a base64 string just like a url and it just returns the data haha! works on safari firefox and chrome so its standard behaviour. We shouldn't return base64 at all though for cached content, that's a lot of video content to convert to base64